chcemvediet/apps/inforequests/admin.py

```diff
@@ -315,3 +317,58 @@ def get_queryset(self, request):
         queryset = queryset.select_related(u'branch')
         queryset = queryset.select_related(u'email')
         return queryset
+
+    def get_actions(self, request):
+        actions = super(ActionAdmin, self).get_actions(request)
+        if u'delete_selected' in actions:
+            del actions[u'delete_selected']
+        return actions
+
+    def has_delete_permission(self, request, obj=None):
+        if obj is None:
+            return True
+        if obj.type in [Action.TYPES.REQUEST, Action.TYPES.ADVANCED_REQUEST]:
+            return False
+        if len(obj.branch.actions) > 1:
+            return True
+        return False
+
+    def render_delete_form(self, request, context):
+        using = router.db_for_write(self.model)
+        collector = admin.utils.NestedObjects(using)
+        collector.collect([context[u'object']])
+        to_delete = collector.nested()
+        for object in traverse(to_delete):
+            if isinstance(object, Action) and object.email:
+                for inforequestemail in object.email.inforequestemail_set.filter(
+                        type=InforequestEmail.TYPES.APPLICANT_ACTION):
+                    context[u'deleted_objects'].append(
+                        format_inforequestemail(inforequestemail, self.admin_site))
+        return super(ActionAdmin, self).render_delete_form(request, context)
+
+    def delete_model(self, request, obj):
+        using = router.db_for_write(self.model)
+        collector = admin.utils.NestedObjects(using)
+        collector.collect([obj])
+        to_delete = collector.nested()
+        for object in traverse(to_delete):
+            if isinstance(object, Action) and object.email:
+                (object.email.inforequestemail_set
+                      .filter(type=InforequestEmail.TYPES.APPLICANT_ACTION).delete())
+                (object.email.inforequestemail_set
+                    .filter(type=InforequestEmail.TYPES.OBLIGEE_ACTION)
+                    .update(type=InforequestEmail.TYPES.UNRELATED))
+        super(ActionAdmin, self).delete_model(request, obj)
+
+def format_inforequestemail(inforequestemail, admin_site):
+    admin_url = reverse(u'{}:inforequests_inforequestemail_change'.format(admin_site.name),
+                        None, (inforequestemail.pk,))
```

Comment:
Ked si pozries, ako pouzivame `reverse` na inych miestach v admine. Tak vsade implicitne predpokladame, ze `admin_site.name` je standardny `admin`. Takze nemusime to pracne vsade zistovat.

In general pri preberani kodu zo vseobecnych libiek, treba orezat a zjednodusit veci, ktore v konkretnej aplikacii riesit nemusime. Kedze mame iba jedno admin rozdhranie, tak nemusime riesit, ktore admin rozhranie sa pouziva a pouzit to jedno.

Reply:
:ok: